### PR TITLE
Address Nils' Comments

### PR DESCRIPTION
**Pull request by ilan-gold**
- Update `event` pages to have call notes and agenda in body with summary in yaml
- use javascript to do dynamic list sorting of dates
- use `summary` from front matter under each call in the list
- add participants page
- Clearly mark zoom dial-in link
- Show file links + descriptions
- local file storage

Screen Shot 2019-11-21 at 3 40 47 PM
Screen Shot 2019-11-21 at 3 40 41 PM
Screen Shot 2019-11-22 at 9 51 30 AM

### CODE REVIEW

File: _events/10.markdown [hunk 1]
```diff
@@ -0,0 +1,9 @@
+---
+layout: event
+title:  "2020-02-11 Call"
+date:   2020-02-11 9:00
```
Comment (mccalluc) [inline _events/10.markdown:4]: Can we avoid requiring the same data to be entered twice?

File: _layouts/event.html [hunk 2]
```diff
@@ -10,27 +10,25 @@
...
     <h2>Files</h2>
-    {%- for page_file in page.file_list -%}
-      •<a href="{{ page_file }}">{{ page_file }}</a> <br>
-    {%- endfor -%}
+    <ul>
+      {%- for page_file in page.file_list -%}
+        {%- if page_file[0] contains '.com' or page_file[0] contains '.edu' or page_file[0] contains '.org' -%}
+          <li><a href="{{ page_file[0] }}">{{ page_file[0] }}</a>: {{page_file[1]}} </li>
+        {% else -%}
+        {% assign file_store = "/file_store/" %}
```
Comment (mccalluc) [inline _layouts/event.html]:
```suggestion
          {% assign file_store = "/file_store/" %}
```

File: _layouts/event.html [hunk 3]
```diff
@@ -10,27 +10,25 @@
...
-    <h2>Presentation</h2>
-    {{ page.presentation }}
+    {%- if page.file_list.size > 0 -%}
     <h2>Files</h2>
-    {%- for page_file in page.file_list -%}
-      •<a href="{{ page_file }}">{{ page_file }}</a> <br>
-    {%- endfor -%}
+    <ul>
+      {%- for page_file in page.file_list -%}
+        {%- if page_file[0] contains '.com' or page_file[0] contains '.edu' or page_file[0] contains '.org' -%}
```
Comment (mccalluc) [inline _layouts/event.html]: No: There are lots of other TLDs. Suggest checking whether the string matches `https?:` as a regex.

File: _layouts/event.html [hunk 4]
```diff
@@ -10,27 +10,25 @@
...
+          <li><a href="{{ page_file[0] }}">{{ page_file[0] }}</a>: {{page_file[1]}} </li>
+        {% else -%}
+        {% assign file_store = "/file_store/" %}
+          <li><a href="{{  file_store | append:  page_file[0] | append: site.baseurl }}">{{ page_file[0] }}</a>: {{page_file[1]}} </li>
```
Comment (mccalluc) [inline _layouts/event.html]: Does this url construction actually work? It's a bit confusing.

File: _layouts/event.html [hunk 5]
```diff
@@ -10,27 +10,25 @@
...
   </header>
 
   <div class="post-content e-content" itemprop="articleBody">
     {{ content }}
-  </div>
-  <h2>Recording</h2>
-    {{ page.recording }}
-    <h2>Presentation</h2>
-    {{ page.presentation }}
+    {%- if page.file_list.size > 0 -%}
```
Comment (mccalluc) [inline _layouts/event.html:19]: I don't remember what counts as truthy here, but `page.file_list` isn't sufficient?

File: _layouts/home.html [hunk 6]
```diff
@@ -12,20 +12,17 @@ <h1>About</h1>
...
-          </a>
-        </h3>
-        {%- if site.show_excerpts -%}
-          {{ event.excerpt }}
-        {%- endif -%}
-      </li>
+      <div id="upcoming-event">
+        {%- for event in site.events -%}
+          <div class="event-link" date="{{ event.date }}">
+              <a  href="{{ event.url | relative_url }}" >{{event.title}}</a>
```
Comment (mccalluc) [inline _layouts/home.html:17]: Why does this need the `relative_url`?

File: _layouts/home.html [hunk 7]
```diff
@@ -12,20 +12,17 @@ <h1>About</h1>
...
-            {{ event.title | escape }}
-          </a>
-        </h3>
-        {%- if site.show_excerpts -%}
-          {{ event.excerpt }}
-        {%- endif -%}
-      </li>
+      <div id="upcoming-event">
+        {%- for event in site.events -%}
+          <div class="event-link" date="{{ event.date }}">
```
Comment (mccalluc) [inline _layouts/home.html]: What uses this? If you need to add ad hoc data, I think `data-` attributes are preferred.

File: assets/list.js [hunk 8]
```diff
@@ -0,0 +1,48 @@
+function makeUL(array) {
+    var list = document.createElement('ul');
+    for (var i = 0; i < array.length; i++) {
```
Comment (mccalluc) [inline assets/list.js]: I'd be comfortable with `array.forEach(function() { })`

Comment (mccalluc) [reply assets/list.js]: particularly since you use `filter` below, which the `=>` syntax.

File: assets/list.js [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+        var item = document.createElement('li');
+        item.appendChild(array[i]);
+        list.appendChild(item);
+    }
+    return list;
+}
+// get list of events
+var eventList = Array.from(document.getElementsByClassName('event-link'))
```
Comment (mccalluc) [inline assets/list.js]: semicolons?

File: assets/list.js [hunk 10]
```diff
@@ -0,0 +1,48 @@
...
+var now = new Date()
+// filter to get upcoming list
+var upcomingList = eventList.filter((a) => new Date(a.attributes.date.value.split(' ')[0] + ' ' + a.attributes.date.value.split(' ')[1]) >= new Date())
```
Comment (mccalluc) [inline assets/list.js]: why not?
```suggestion
var upcomingList = eventList.filter((a) => new Date(a.attributes.date.value) >= new Date())
```

File: assets/list.js [hunk 11]
```diff
@@ -0,0 +1,48 @@
...
+var eventDiv = document.getElementById('event-list')
+var upcomingEventDiv = document.getElementById('upcoming-event')
+// this is the events page
```
Comment (mccalluc) [inline assets/list.js]: Slight (nit-picky) preference to capitalize, and have some closing punctuation (':', or '.', or '...')

File: assets/list.js [hunk 12]
```diff
@@ -0,0 +1,48 @@
...
+  pastDiv.id = 'past'
+  pastDiv.appendChild(makeUL(pastList))
+  var pastHeader = document.createElement('H3')
+  text = document.createTextNode("Past Calls");
+  pastHeader.appendChild(text)
+  eventDiv.appendChild(pastHeader)
+  eventDiv.appendChild(pastDiv)
+}
+// this is the home page
+else if (upcomingEventDiv != undefined) {
```
Comment (mccalluc) [inline assets/list.js]: Generally prefer putting the shorter case first in an if-then ("guard clause")... though it might make the boolean more complicated?

File: events.markdown [hunk 13]
```diff
@@ -2,21 +2,15 @@
...
+{%- for event in site.events -%}
+<div class="event-link" date="{{ event.date }}">
+<a  href="{{ event.url | relative_url }}" >{{event.title}}</a>
 <br/>
-<hr/>
+{{event.summary}}
 <br/>
-### Past Events
 <br/>
-{%- assign sorted = (site.events | where_exp: "item", "item.date < site.time" -%}
```
Comment (mccalluc) [inline events.markdown:18]: Why did you need to give on this and move the logic into JS?

File: participants.markdown [hunk 14]
```diff
@@ -0,0 +1,6 @@
+---
+layout: page
+title: Participants
+---
+## [Nils Gehlenborg](gehlenborglab.com)
```
Comment (mccalluc) [inline participants.markdown]:
```suggestion
## [Nils Gehlenborg](http://gehlenborglab.com)
```
Probably going to be a relative url without that.

File: events.markdown [hunk 15]
```diff
@@ -2,21 +2,15 @@
...
-{%- for event in sorted -%}
-  [**{{event.title}}**]({{ event.url | relative_url }})
-  {{event.content}}
+</div>                                   
 {%- endfor -%}
+</div>
```
Comment (mccalluc) [inline events.markdown:30]: Somewhat concerned about file_store... Github really isn't the place for large binary files.

**Review by mccalluc — CHANGES_REQUESTED** [review]
Some qualms, but you know more about project direction than I do.

File: _events/10.markdown (hunk 1 above)
Comment (ilan-gold) [reply _events/10.markdown:4]: I think @ngehlenborg should weigh in here if there will be a title that is more than just the date of the call (that title was just the placeholder I put in).  I wanted to leave it as an option but could also see it as unnecessary.

File: _layouts/event.html (hunk 5 above)
Comment (ilan-gold) [reply _layouts/event.html:19]: My experience was that it was not sufficient

File: assets/list.js (hunk 10 above)
Comment (ilan-gold) [reply assets/list.js]: the date format from jekyll was not playing with the `Date()` cast

File: events.markdown (hunk 13 above)
Comment (ilan-gold) [reply events.markdown:18]: As I understand it, once the site is built, the list is fixed and since these rely on the current date, we need it to be more dynamic

File: events.markdown (hunk 15 above)
Comment (ilan-gold) [reply events.markdown:30]: @ngehlenborg requested that pdf's could be stored here but that is why I left it open-ended to links, so that we have options.

File: _layouts/home.html (hunk 7 above)
Comment (ilan-gold) [reply _layouts/home.html]: This is specifically for the Javavscript to access dynamically

File: _layouts/event.html (hunk 4 above)
Comment (ilan-gold) [reply _layouts/event.html]: as far as i can tell, it does

File: _layouts/home.html (hunk 6 above)
Comment (ilan-gold) [reply _layouts/home.html:17]: the Github Pages needs to use this since its not at the root i.e `hms-dbmi.github.io/x-consortium/xyz/foo`

File: _layouts/event.html (hunk 3 above)
Comment (ilan-gold) [reply _layouts/event.html]: I don't think Liquid provides regex checking but a `contains` should do fine, I think.

Comment (mccalluc) [reply _layouts/event.html]: @ilan-gold : Sorry, my point is that you probably want to handle `http://www.embl.de` like you would `http://harvard.edu`: It's the protocol rather than the tld that you should look for.

File: _layouts/event.html (hunk 4 above)
Comment (mccalluc) [reply _layouts/event.html]: I guess I'm not convinced the append makes things clearer... I might be missing something?
```
{{  file_store }}{{ page_file[0] }}{{  site.baseurl }}
```

File: _layouts/home.html (hunk 7 above)
Comment (mccalluc) [reply _layouts/home.html]: Right, but if you're adding ad-hoc data to an element, it should be in a `data-` attribute: https://developer.mozilla.org/en-US/docs/Learn/HTML/Howto/Use_data_attributes

File: assets/list.js [hunk 16]
```diff
@@ -0,0 +1,21 @@
...
+// This is the events page.
+else if(eventLists != undefined){
+  $('#upcoming [data-date]').filter(function(){
+      return (new Date($(this).data('date').split(' ')[0] + ' ' + $(this).data('date').split(' ')[1]) < now);
+    }).remove();
+  $('#past [data-date]').filter(function(){
+      return (new Date($(this).data('date').split(' ')[0] + ' ' + $(this).data('date').split(' ')[1]) >= now);
+    }).remove();
+  $('#past').children().each((e,i) => $('#past').prepend(i))
+}
```
Comment (mccalluc) [inline assets/list.js]: Define a function rather than copy-and-pasting:
```javascript
function asDate(el) {
  return new Date($(this).data('date').split(' ')[0] +  ' ' + $(this).data('date').split(' ')[1])
}
```

File: assets/list.js [hunk 17]
```diff
@@ -0,0 +1,21 @@
+var now = new Date();
+// Filter to get upcoming list.
+var eventLists = document.getElementById('event-lists');
+var homeEvents = document.getElementById('home-events');
+// This is the home page.
+if (homeEvents != undefined) {
```
Comment (mccalluc) [inline assets/list.js]: I don't think the these checks are needed now: the jquery will only do something when the pattern matches.

File: events.markdown (hunk 15 above)
Comment (mccalluc) [reply events.markdown:30]: okey-dokey!

**Review by mccalluc — COMMENTED** [review]
A few more tweaks. Thanks!
